File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode p = head;
        ListNode q = head;
        for(int i = 1; i <= n; i++){
            p = p.next;
        }
        if(p == null){
            return head.next;
        }
        while(p.next != null){
            p = p.next;
            q = q.next;
        }
        q.next = q.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        p = head
        q = head
        for i in range(0,n):
            p = p.next
        if not p:
            return head.next
        while p.next is not None:
            p = p.next
            q = q.next
        q.next = q.next.next
        return head
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        